editoast/src/generated_data/error/catenaries.rs

```diff
@@ -59,10 +67,51 @@ pub fn check_catenary_track_ranges(
     infra_errors
 }
 
+/// Checks that speed sections overlapping
+pub fn check_overlapping(infra_cache: &InfraCache, _: &Graph) -> Vec<InfraError> {
+    let mut overlapping_sc = HashSet::new();
+    // Key: (Track, Tag, Direction)
+    //   - Tag is None for the default speed limit and Some for the tagged ones
+    // Value: RangeMap<Range, SpeedSectionId>
+    let mut range_maps: HashMap<(String, Option<String>, Direction), RangeMap<u64, String>> =
+        Default::default();
+
+    // Iterate over all the speed sections insure we don't report duplicated errors
+    for catenary in infra_cache.catenaries().values() {
+        let catenary = catenary.unwrap_catenary();
+
+        for track_range in catenary.track_ranges.iter() {
+            let range = (track_range.begin * 100.) as u64..(track_range.end * 100.) as u64;
+            let track_id = &track_range.track.0;
+
+            for direction in get_directions(track_range.applicable_directions) {
```

Comment:
You can ignore directions for catenaries, they'll be removed from the schemas at some point anyway